### PR TITLE
Organize SendReceive tests and isolate non-parallel test collection

### LINKED ISSUE #1712: Test: System.Net.Sockets.Tests.SendReceiveEap.SendToRecvFrom_Datagram_UDP(loopbackAddress: ::1) failed in CI
Failed test: System.Net.Sockets.Tests.SendReceiveEap.SendToRecvFrom_Datagram_UDP(loopbackAddress: ::1)
Configuration: OuterLoop_CentOS7.1_debug ([build#123](https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_centos7.1_debug/123/testReport/))

Message:
~~~
System.Net.Sockets.Tests.SendReceiveEap.SendToRecvFrom_Datagram_UDP(loopbackAddress: ::1) [FAIL]
        Assert.True() Failure
        Expected: True
        Actual:   False
~~~

Stack Trace:
~~~
/mnt/resource/j/workspace/dotnet_corefx/master/outerloop_centos7.1_debug/src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs(148,0): at System.Net.Sockets.Tests.SendReceive.<SendToRecvFrom_Datagram_UDP>d__17.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
~~~

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_centos7.1_debug/123/consoleText

### PR DESCRIPTION
**Pull request by antonfirsov**
Some `SendReceive` socket tests may be prone to timing issues on CI. This seems to be the root cause of #1712. We need a reliable way to run such tests to unblock the work on new UDP socket API-s in #33418.

This PR defines a new `SendReceiveNonParallel` test group, moving `SendToRecvFrom_Datagram_UDP` into that group. Since this is already a significant reorganization, it seemed reasonable to also:
- Harmonize naming: all SendReceive test classses are now named either  `SendReceive_[SubVariant]` 
 or `SendReceiveNonParallel_[SubVariant]`
- Split `SendReceive.cs` into multiple files:
    - `SendReceive.cs` for the parallel variants
    - `SendReceiveNonParallel.cs` for the new, non-parallel variants
    - Rename the non-generic class `SendReceive` to `SendReceiveCommon` (to avoid name collision and confusion with `SendReceive<T>`) and move it to `SendReceiveCommon.cs` 
    - Move `SendReceiveListener` and `SendReceiveUdpClient` to separate files

Fixes #1712 

/cc @geoffkizer

### CODE REVIEW

File: src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceiveCommon.cs [hunk 1]
```diff
@@ -0,0 +1,286 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
```
Comment (geoffkizer) [inline src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceiveCommon.cs:4]: "Common" is kind of a strange term for this... it seems like it's really just a handful of send/receive related tests that don't fit elsewhere. I'd suggest calling it "SendReceiveMisc.cs" or something like that.

File: src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceiveListener.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
```
Comment (geoffkizer) [inline src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceiveListener.cs:4]: I'd call this file SendReceiveTcpClient to match SendReceiveUdpClient.

**Review by geoffkizer — CHANGES_REQUESTED** [review]
Generally looks good, but I think we should change a couple class/file names... see above.

**Review by geoffkizer — APPROVED** [review]

**Comment by antonfirsov** [comment]
The OuterLoop test failures are unrelated (System.Xml, Http and #44015).

**Pull request merged**